### PR TITLE
Add verification of proper setup of build runner

### PR DESCRIPTION
**Pull request by KrauseFx**
This will save us debugging time in the future

### CODE REVIEW

File: app/features/build_runner/fastlane_build_runner.rb [hunk 1]
```diff
@@ -39,6 +39,10 @@ def setup(parameters: nil)
 
     # completion_block is called with an array of artifacts
     def run(new_line_block:, completion_block:)
+      if lane.nil?
+        raise "Before calling `.run` on #{self}, you have to call `.setup` to finish preparing the BuildRunner"
```
Comment (taquitos) [inline app/features/build_runner/fastlane_build_runner.rb]: Is there anything we should do the with the `completion_block`? Does anything rely on it being called?

Comment (KrauseFx) [reply app/features/build_runner/fastlane_build_runner.rb]: `completion_block` is already a required parameter it seems? Note that we check if `self.lane` was defined, which is set as part of `setup`, so we can ensure the setup was properly executed.

Comment (KrauseFx) [reply app/features/build_runner/fastlane_build_runner.rb]: Also `run` is called from the parent class only, not from the outside world it seems, coming from calling `start` from the `BuildRunner` object

**Comment by KrauseFx** [comment]
Ready to be merged from my side 👍

**Comment by KrauseFx** [comment]
Gonna trigger that build again

**Comment by AndrewMcBurney** [comment]
Hey @KrauseFx, I reran the build manually, and the checks are ✅. I think we're good to :shipit:

**Comment by KrauseFx** [comment]
Beautiful, thanks @AndrewMcBurney 👍

**Pull request merged**